vertical-pod-autoscaler/pkg/recommender/input/metrics/metrics_custom_lister.go

```diff
@@ -110,15 +93,20 @@ func (c *customPodMetricsLister) List(ctx context.Context, namespace string, opt
 
 		go func(query nsQuery) {
 			defer wg.Done()
-			resChan <- c.query(query)
+
+			// Time out the query after 45 seconds.
+			queryCtx, cancel := context.WithTimeout(ctx, 45*time.Second)
```

Comment:
We need to signal that requests are timing out. I think we should do two things:
1) Emit error log with some details e.g. for which namespace the request timed out.
2) Emit metric so that we can set alert on it.

Reply:
There is an error log [here](https://github.com/jodzga/autoscaler/blob/ff08bcf58ff44e8fae1b9fa60a0ed336299523ac/vertical-pod-autoscaler/pkg/recommender/input/metrics/metrics_custom_lister.go#L124-L125) that will capture this, and I have a TODO there to add metrics. There are multiple other places to add metrics and I'd prefer to do those in a single batch.

Reply:
SG, thanks!